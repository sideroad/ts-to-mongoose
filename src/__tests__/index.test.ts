import exec from '../index';
import path from 'path';
test('generate mongoose schema definition from typescript', () => {
  expect(
    exec({ file: path.join(__dirname, '../../examples/ts/sample.ts') })
  ).toMatchSnapshot();
});
