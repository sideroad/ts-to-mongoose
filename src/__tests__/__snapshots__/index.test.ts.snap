// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate mongoose schema definition from typescript 1`] = `
"export const Extendee = { extended: { type: String } };

export const Type = { type: String };

export const ArrayedObject = { a: { type: String }, b: { type: Number } };

export const Sample = {
  string: { type: String },
  number: { type: Number },
  boolean: { type: Boolean },
  date: { type: Date },
  stringArray: [ { type: String } ],
  objectArray: [
    {
      a: { type: String },
      b: { type: Number }
    }
  ],
  enums: { enum: [ 'a', 'b' ], type: String },
  type: { type: String },
  required: { type: String, required: true },
  id: { type: String, required: true, unique: true },
  extended: { type: String }
};
"
`;
